fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductAnalysis.kt

```diff
@@ -0,0 +1,158 @@
+ *
+ * @param productRecommendationsPreference The current state of the product recommendations toggle.
+ * @param productAnalysis The product analysis to display.
+ * @param onOptOutClick Invoked when the user opts out of the review quality check feature.
+ * @param onProductRecommendationsEnabledStateChange Invoked when the user changes the product
+ * recommendations toggle state.
+ * @param modifier The modifier to be applied to the Composable.
+ */
+@Composable
+fun ProductAnalysis(
+    productRecommendationsPreference: Boolean,
+    productAnalysis: ProductAnalysis,
+    onOptOutClick: () -> Unit,
+    onProductRecommendationsEnabledStateChange: (Boolean) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Column(
+        modifier = modifier,
+        verticalArrangement = Arrangement.spacedBy(16.dp),
+    ) {
+        ReviewGradeCard(
+            reviewGrade = productAnalysis.reviewGrade,
+            modifier = Modifier.fillMaxWidth(),
+        )
+
+        SettingsCard(
+            modifier = Modifier.fillMaxWidth(),
+            productRecommendationsEnabled = productRecommendationsPreference,
+            onProductRecommendationsEnabledStateChange = onProductRecommendationsEnabledStateChange,
+            onTurnOffReviewQualityCheckClick = onOptOutClick,
+        )
+    }
+}
+
+@Composable
+private fun ReviewGradeCard(
+    reviewGrade: ReviewQualityCheckState.Grade,
+    modifier: Modifier = Modifier,
+) {
+    ReviewQualityCheckCard(modifier = modifier.semantics(mergeDescendants = true) {}) {
+        Text(
+            text = stringResource(R.string.review_quality_check_grade_title),
+            color = FirefoxTheme.colors.textPrimary,
+            style = FirefoxTheme.typography.headline8,
+        )
+
+        Spacer(modifier = Modifier.height(8.dp))
+
+        ReviewGradeExpanded(grade = reviewGrade)
+    }
+}
+
+@Composable
+private fun SettingsCard(
+    productRecommendationsEnabled: Boolean,
+    onProductRecommendationsEnabledStateChange: (Boolean) -> Unit,
+    onTurnOffReviewQualityCheckClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    ReviewQualityCheckExpandableCard(
+        modifier = modifier,
+        title = stringResource(R.string.review_quality_check_settings_title),
+    ) {
+        Column {
+            Spacer(modifier = Modifier.height(8.dp))
+
+            SwitchWithLabel(
+                checked = productRecommendationsEnabled,
+                onCheckedChange = onProductRecommendationsEnabledStateChange,
+                label = stringResource(R.string.review_quality_check_settings_recommended_products),
+            )
+
+            Spacer(modifier = Modifier.height(16.dp))
+
+            SecondaryButton(
+                text = stringResource(R.string.review_quality_check_settings_turn_off),
+                onClick = onTurnOffReviewQualityCheckClick,
+            )
+        }
+    }
+}
+
+@Composable
+@LightDarkPreview
+private fun ProductAnalysisPreview() {
```

Comment:
nit: would it be possible to have the settings toggle interactable? Either here or in `ReviewQualityCheckBottomSheet`?

Reply:
sure thing, it is now in this preview.